index.bs

```diff
@@ -2723,6 +2770,233 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
+
+1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+   with |context id|.
+
+1. Assert: |context| is not null.
+
+1. Let |maximum returned nodes| be the value of the <code>maxNodes</code>
+   field of |command parameters|, if present, or 0 otherwise.
+
+1. Let |current context target| be the a [=/map=] matching of the
+   <code>script.ContextTarget</code> production with the 
+   <code>context</code> field set to the [=browsing context id=]
+   of |context|.
+
+1. Let |realm| be the result of [=trying=] to [=get a realm from a target=]
+   given |current context target|.
+
+1. Let |locator| be the value of the <code>locator</code> field
+   of |command parameters|.
+
+1. Assert: |locator| is not null.
+
+1. Let |start nodes parameter| be the value of the <code>startNodes</code> field
+   of |command parameters|.
+
+1. Assert: |start nodes parameter| is not null.
+
+1. Let |context nodes| be an empty array.
+
+1. For each |serialized start node| in |start nodes parameter|:
+
+   1. Let |start node| be the result of [=trying=] to [=deserialize shared reference=] given
+      |serialized start node|, |realm| and |session|.
+
+   1. Append |start node| to |context nodes|.
+
+1. If the <code>length</code> property of |context nodes| is 0, append the node 
+   corresponding to the <code>documentElement</code> property of |context|'s
+   [=active document=] to |context nodes|.
+
+1. Let |result nodes| be an empty array.
+
+1. Let |type| be the value of the <code>type</code> property of |locator|.
+
+1. In the following list of conditions and associated steps, run the first set 
+   of steps for which the associated condition is true, if any:
+
+   <dl>
+
+      <dt>|type| is the string "<code>css</code>"
+      <dd>
+         1. Let |selector| be the value of the <code>value</code> property of |locator|.
+
+         1. Let |result nodes| be a result of [=trying=] to [=locate nodes using css=]
+            given |current context target|, |context nodes|, |selector|, |maximum returned
+            nodes| and |session|.
+
+      <dt>|type| is the string "<code>xpath</code>"
+      <dd>
+         1. Let |selector| be the value of the <code>value</code> property of |locator|.
+
+         1. Let |result nodes| be a result of [=trying=] to [=locate nodes using xpath=]
+            given |current context target|, |context nodes|, |selector|, |maximum returned
+            nodes| and |session|.
+
+      <dt>|type| is the string "<code>text</code>"
+      <dd>
+         1. Let |selector| be the value of the <code>value</code> property of |locator|.
+
+         1. Let |regexp| be the value of the <code>data</code> property of |selector|.
+
+         1. Let |result nodes| be a result of [=trying=] to [=locate nodes using regexp=]
+            given |current context target|, |context nodes|, |regexp|, |maximum returned
+            nodes| and |session|.
+
+1. If |maximum returned nodes| is greater than 0, truncate |result nodes| to the
```

Comment:
Since each algorithm is already truncating, I don't think that's possible? It could be an assert.

Reply:
You're right. That was a recent change made yesterday, and I forgot to update this step. Will fix.